columnflow/production/categories.py

```diff
@@ -65,33 +60,32 @@ def category_ids(
 
 @category_ids.init
 def category_ids_init(self: Producer) -> None:
-    # store a mapping from leaf category to selector classes for faster lookup
-    self.category_to_selectors = defaultdict(list)
+    # store a mapping from leaf category to categorizer classes for faster lookup
+    self.categorizer_map = defaultdict(list)
 
-    # add all selectors obtained from leaf category selection expressions to the used columns
+    # add all categorizers obtained from leaf category selection expressions to the used columns
     for cat_inst in self.config_inst.get_leaf_categories():
-        # treat all selections as lists
+        # treat all selections as lists of categorizers
         for sel in law.util.make_list(cat_inst.selection):
-            if Selector.derived_by(sel):
-                selector = sel
-            elif Selector.has_cls(sel):
-                selector = Selector.get_cls(sel)
+            if Categorizer.derived_by(sel):
+                categorizer = sel
+            elif Categorizer.has_cls(sel):
+                categorizer = Categorizer.get_cls(sel)
             else:
                 raise Exception(
                     f"selection '{sel}' of category '{cat_inst.name}' cannot be resolved to an "
-                    "existing Selector object",
+                    "existing Categorizer object",
                 )
 
-            # variables should refer to unexposed selectors as they should usually not
-            # return SelectionResult's but a flat per-event mask
-            if selector.exposed:
-                logger.warning(
-                    f"selection of category {cat_inst.name} seems to refer to an exposed selector "
-                    "whose return value is most likely incompatible with category masks",
+            # the categorizer must be exposed
+            if not categorizer.exposed:
+                raise RuntimeError(
+                    f"cannot use unexposed categorizer '{categorizer}' to evaluate category "
+                    f"{cat_inst}",
                 )
```

Comment:
Just for clarification, this means that we also plan to support hidden categorizers, but what's the use case for that? Sth like a "super" categorizer that consists of smaller ones? Not opposed to that idea, but since they only provide boolean masks and can't easily provide additional information (like selectors can), I don't think this is strictly necessary, right?

Reply:
In general, I think it's good to have the option to mark a selector / producer / calibrator as "not exposed", i.e., an error will show up when someone tries to use them from the command line.

My thought behind adding the same for categorizers was a) consistency, and b) although they're not used from the command line, but by `od.Category` objects, it could be useful to mark certain categorizers as "do not use in categories, but call from another function". This could be the case when a categorizer has additional arguments, that cannot be automatically known by `od.Category`'s.

Reply:
OK sounds good to me :+1: